src/id.rs

```diff
@@ -53,10 +56,11 @@ impl Id {
 
     pub fn with_keys(sign_keys: (crypto::sign::PublicKey, crypto::sign::SecretKey),
                      encrypt_keys: (crypto::box_::PublicKey, crypto::box_::SecretKey))-> Id {
+        let name = NameType::new(crypto::hash::sha512::hash(&sign_keys.0[..]).0); 
```

Comment:
Could we use this function?
https://github.com/maidsafe/routing/blob/master/src/utils.rs#L42
Or does that one have different semantic?